src/Microsoft.Diagnostics.EventFlow.Inputs.Trace/TraceInput.cs

```diff
@@ -81,6 +81,36 @@ public override void Fail(string message, string detailMessage)
             SubmitEventData(message, TraceEventType.Error);
         }
 
+        public override void Write(object o)
+        {
+            WriteLine(o.ToString());
+        }
+
+        public override void Write(object o, string category)
```

Comment:
For this, and other overrides that take the `category` parameter, we should not lose the category information. This probably means adding another parameter to `SubmitEventData` so that the category information can be used to populate `EventData` object. For example:

```csharp
        private void SubmitEventData(string message, TraceEventType level, int? id = null, string source = null, string relatedActivityId = null, string category = null)
        {
            try
            {
                EventData eventEntry = new EventData()
                {
                    ProviderName = string.IsNullOrEmpty(source) ? TraceTag : source,
                    Timestamp = DateTimePrecise.UtcNow,
                    Level = ToEventLevel(level)
                };

                var eventPayload = eventEntry.Payload;
                eventPayload["Message"] = message;

                if (id != null && id.HasValue)
                {
                    eventPayload["EventId"] = id.Value;
                }

                if (!string.IsNullOrEmpty(relatedActivityId))
                {
                    eventPayload["RelatedActivityID"] = relatedActivityId;
                }

                if (!string.IsNullOrEmpty (category))
                {
                    eventPayload["Category"] = category;
                }

                // (rest of the SubmitEventData() code unchanged)
```

Reply:
This implies for these overloads we would not rely on `WriteLine()`, but call `SubmitEventData()` directly.